packages/main/src/SegmentedButton.ts

```diff
@@ -196,24 +224,35 @@ class SegmentedButton extends UI5Element {
 			return;
 		}
 
-		if (target !== this._selectedItem) {
-			if (this._selectedItem) {
-				this._selectedItem.pressed = false;
+		switch (this.mode) {
+		case SegmentedButtonMode.Single:
```

Comment:
As the "Single" is the default mode, I would structure the code here differently although the result should be the same

case: Multi -> _applyMultiSelection(target)
case: Default -> _applySingleSelectionTo(target);

Reply:
Done